generate_rospkg_apkbuild/APKBUILD.em.sh

```diff
@@ -124,12 +152,34 @@ check() {
     --catkin-make-args run_tests 2>&1 | tee $checklog
   catkin_test_results 2>&1 | tee $checklog
 @[  end if]@
-@[  if use_cmake]@
-  cd src/$_pkgname/build
+@[if (use_ament_cmake or use_ament_python) and ros2_workspace_available]@
+  source /usr/ros/@(ros_distro)/setup.sh
+@[end if]@
+@[if use_ament_cmake or use_ament_python]@
+  export PYTHONPATH="$pkgdir"/usr/ros/@(ros_distro)/lib/python$(python3 -V | sed -e "s/\(Python\s\)\(\d\.\d*\)\(\..*\)/\2/")/site-packages:${PYTHONPATH}
+  export AMENT_PREFIX_PATH="$pkgdir"/usr/ros/@(ros_distro):${AMENT_PREFIX_PATH}
+  export PATH="$pkgdir"/usr/ros/@(ros_distro)/bin:${PATH}
+@[end if]@
+@[  if use_cmake or use_ament_cmake]@
+  cd build
   if [ $(make -q test > /dev/null 2> /dev/null; echo $?) -eq 1 ]; then
     make test 2>&1 | tee $checklog
   fi
 @[  end if]@
+@[if use_ament_python]@
+  cd src/$_pkgname
+  TEST_TARGET=$(ls -d */ | grep -m1 "\(test\|tests\)") || true
+  if [ -z "$TEST_TARGET" ]; then
+    echo "No \"test\" or \"tests\" directory. Check skipped" | tee $checklog
+    return 0
+  fi
+  USE_PYTEST=$(grep pytest setup.py) || true
+  if [ -n "$USE_PYTEST" ]; then
+    python -m pytest 2>&1 | tee $checklog
+  else
+    python setup.py test 2>&1 | tee $checklog
```

Comment:
Are there standardized ways to run tests in ament_python?

Reply:
https://colcon.readthedocs.io/en/released/reference/verb/test.html
Colcon determines which test to use by checking the dependencies defined in `setup.py` https://github.com/colcon/colcon-core/blob/a6c04e04b3b6494c75d660eb9d5fa043e6cd331a/colcon_core/task/python/test/pytest.py#L50 .

This is not the same way as colcon but I think it's not matter for most cases (unless there is an ament_python package whose `setup.py` contains the string `pytest` but `pytest` can't be used to test the package).